pkg/server/server.go

```diff
@@ -914,9 +930,14 @@ func (s *Server) Read(ctx context.Context, req *openfgav1.ReadRequest) (*openfga
 
 	ctx = telemetry.ContextWithRPCInfo(ctx, telemetry.RPCInfo{
 		Service: s.serviceName,
-		Method:  "Read",
+		Method:  authz.Read,
 	})
 
+	err = s.CheckAuthz(ctx, req.GetStoreId(), authz.Read)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is a code smell having to add CheckAuthz to every endpoint here, because when we add a new endpoint we will have to remember to add this line.

Is there anything preventing you from writing this PR as an authorization middleware? Could you tell us what they are?

I prefer addressing those missing pieces first and then coming back to this PR.

cc @openfga/backend

Reply:
Agree with Maria here, it's not only a code smell but a security liability.

Reply:
Discussed offline, we can't do as a middleware because people using OpenFGA as a library and not a service will miss out on this feature (since there are no middlewares when using a library)

Reply:
@aaguiarz what about Okta FGA?